experimental/CliffordAlgebras/src/CliffordAlgebras.jl

```diff
@@ -0,0 +1,584 @@
+  disq
+
+################################################################################
+#
+#  Datatypes
+#
+################################################################################
+
+### Algebra ###
+mutable struct CliffordAlgebra{T,S} <: Hecke.AbstractAssociativeAlgebra{T}
+  base_ring::Ring
+  space::Hecke.QuadSpace{K,S} where {K}
+  gram::S
+  dim::Int
+  centroid::Any
+  disq::T
+  center::Any
+
+  #Return the Clifford algebra of the quadratic space 'qs' 
+  function CliffordAlgebra{T,S}(qs::Hecke.QuadSpace{K,S}) where {T,S,K}
+    gram = gram_matrix(qs)
+    return new{T,S}(base_ring(qs), qs, gram, 2^ncols(gram))
+  end
+end
+
+### Elements ###
+mutable struct CliffordAlgebraElem{T,S,P} <: Hecke.AbstractAssociativeAlgebraElem{T}
+  parent::P
+  coeffs::Vector{T}
+  even_coeffs::Vector{T}
+  odd_coeffs::Vector{T}
+
+  #Return the 0-element of the Clifford algebra C
+  function CliffordAlgebraElem{T,S,P}(C::CliffordAlgebra{T,S}) where {T,S,P}
+    newelt = new{T,S,CliffordAlgebra{T,S}}(C, fill(C.base_ring(), C.dim))
+    _set_even_odd_coefficients!(newelt)
+    return newelt
+  end
+
+  CliffordAlgebraElem(C::CliffordAlgebra) =
+    CliffordAlgebraElem{elem_type(C.base_ring),typeof(C.gram),typeof(C)}(C)
+
+  #Return the element in the Clifford algebra C with coefficient vector coeff wrt. the canonical basis
+  function CliffordAlgebraElem{T,S,P}(
+    C::CliffordAlgebra{T,S}, coeff::Vector{R}
+  ) where {T,S,P,R<:FieldElem}
+    @req length(coeff) == C.dim "invalid length of coefficient vector"
+    newelt = new{T,S,CliffordAlgebra{T,S}}(C, coeff)
+    _set_even_odd_coefficients!(newelt)
+    return newelt
+  end
+
+  CliffordAlgebraElem(C::CliffordAlgebra{T,S}, coeff::Vector{T}) where {T,S} =
+    CliffordAlgebraElem{elem_type(C.base_ring),typeof(C.gram),typeof(C)}(C, coeff)
+
+  function CliffordAlgebraElem(C::CliffordAlgebra{T,S}, coeff::Vector{R}) where {T,S,R}
+    K = C.base_ring
+    @req __can_convert_coefficients(coeff, K) "entries of coefficient vector are not contained in $(K)"
+    return CliffordAlgebraElem{elem_type(C.base_ring),typeof(C.gram),typeof(C)}(
+      C, K.(coeff)
+    )
+  end
+end
+
+elem_type(::Type{CliffordAlgebra{T,S}}) where {T,S} =
+  CliffordAlgebraElem{T,S,CliffordAlgebra{T,S}}
+
+parent_type(::Type{CliffordAlgebraElem{T,S,P}}) where {T,S,P} = P
+
+base_ring_type(C::CliffordAlgebra{T,S}) where {T,S} = typeof(base_ring(C))
+
+################################################################################
+#
+#  Construction
+#
+################################################################################
+
+### Algebra ###
+
+@doc raw"""
+    clifford_algebra(qs::QuadSpace) -> CliffordAlgebra
+
+Return the Clifford algebra of the quadratic space 'qs'.
+"""
+clifford_algebra(qs::Hecke.QuadSpace) =
+  CliffordAlgebra{elem_type(base_ring(qs)),typeof(gram_matrix(qs))}(qs)
+
+### Elements ###
+(C::CliffordAlgebra)() = CliffordAlgebraElem(C)
+
+function (C::CliffordAlgebra)(a::R) where {R<:RingElem}
```

Comment:
```suggestion
function (C::CliffordAlgebra)(a::R) where {R<:RingElement}
```
See
```
julia> RingElement
RingElement (alias for Union{RingElem, AbstractFloat, Integer, Rational})
```
Maybe this includes all the things you want and you can remove the `Number` version below?

Reply:
I have tried this in the past but it doesn't work. In several tests I get:

MethodError: (::CliffordAlgebra{AbsSimpleNumFieldElem, AbstractAlgebra.Generic.MatSpaceElem{AbsSimpleNumFieldElem}})(::AbsSimpleNumFieldElem) is ambiguous

MethodError: (::CliffordAlgebra{QQFieldElem, QQMatrix})(::QQFieldElem) is ambiguous